pytket_qir/converter.py

```diff
@@ -0,0 +1,357 @@
+
+            # import pdb; pdb.set_trace()
+            if block_name == "entry":
+                # self.conditions["entry_pred"] = True
+                # self.local_conditions["entry_pred"] = True
+                self.edges[block_name] = {block_name: True}
+            if isinstance(term, QirCondBrTerminator):
+                f = lambda el: el.name == term.true_dest or el.name == term.false_dest
+                self._successors[block_name] = list(
+                    map(lambda el: el.name, filter(f, self.module.functions[0].blocks))
+                )
+                self.edges[block_name] = {
+                    term.true_dest: True,
+                    term.false_dest: False,
+                }
+            elif isinstance(term, QirBrTerminator):
+                self._successors[block_name] = [term.dest]
+                self.edges[block_name] = {term.dest: None}
+            elif isinstance(term, QirRetTerminator):
+                self._successors[block_name] = []
+                self.edges[block_name] = {}
+        # Sort the DAG.
+        sorted_dag = topological_sort(self._successors)
+        # Update (sort) the list of successors.
+        for block, succs in sorted_dag.items():
+            if len(succs) > 1:
+                f = (
+                    lambda el: el == self._successors[block][0]
+                    or el == self._successors[block][1]
+                )
+                self._successors[block] = list(filter(f, self._successors))
+
+    @property
+    def predecessors(self):
+        return self._predecessors
+
+    @predecessors.setter
+    def predecessors(self, value):
+        self._predecessors = OrderedDict()
+        reversed_succs = reversed(self.successors)
+        reversed_succs_list = list(reversed_succs)
+        for curr_block_index, curr_block_name in enumerate(reversed_succs_list):
+            predecessors = []
+            for next_block_name in reversed_succs_list[curr_block_index + 1 :]:
+                if curr_block_name in self.successors[next_block_name]:
+                    predecessors.append(next_block_name)
+            self._predecessors[curr_block_name] = predecessors
+
+    @property
+    def conditions(self):
+        return self._conditions
+
+    @conditions.setter
+    def conditions(self, value):
+        self._conditions = value
+
+    @property
+    def edges(self):
+        return self._edges
+
+    @edges.setter
+    def edges(self, value):
+        self._edges = value
+
+    @property
+    def local_conditions(self):
+        return self._local_conditions
+
+    @local_conditions.setter
+    def local_conditions(self, value):
+        self._local_conditions = value
+
+    @property
+    def cfg(self):
+        return self._cfg
+
+    @cfg.setter
+    def cfg(self, value) -> None:
+        # import pdb; pdb.set_trace()
+        self.conditions = {}
+        self.edges = {}
+        self.local_conditions = {}
+        self.successors = {}
```

Comment:
This will call the function from above? can you maybe add a comment to point this out?

Reply:
Done.